diff --git a/CMakeLists.txt b/CMakeLists.txt
index 25161385f..90468988a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -148 +148,5 @@ if (GNU_COMPILER OR CLANG_COMPILER)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -Wall -pedantic -Wextra")
+        if (EMSCRIPTEN)
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pedantic -Wextra")
+        else ()
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -Wall -pedantic -Wextra")
+        endif ()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fc734a649..074c93805 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -118,0 +119,4 @@ export(EXPORT SUMOConfig NAMESPACE SUMO:: FILE "${CMAKE_BINARY_DIR}/SUMOConfig.cmake")
+if (SIMFORGE_SUMO_WASM_TARGET)
+    include("${SIMFORGE_SUMO_WASM_TARGET}")
+endif ()
+
diff --git a/src/utils/iodevices/OutputDevice_File.cpp b/src/utils/iodevices/OutputDevice_File.cpp
index 6cbaef6c8..4e36fd0e4 100644
--- a/src/utils/iodevices/OutputDevice_File.cpp
+++ b/src/utils/iodevices/OutputDevice_File.cpp
@@ -24,0 +25 @@
+#include <fstream>
@@ -67,2 +67,0 @@ OutputDevice_File::OutputDevice_File(const std::string& fullName, const bool binary, const bool compressed)
-#else
-    UNUSED_PARAMETER(compressed);
